src/Handler/Workspace/Drop/DropWorkspaceHandler.php

```
@@ -0,0 +1,64 @@
+use Keboola\StorageDriver\Command\Workspace\DropWorkspaceCommand;
+use Keboola\StorageDriver\Contract\Driver\Command\DriverCommandHandlerInterface;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\Teradata\TeradataSessionManager;
+use Keboola\TableBackendUtils\Escaping\Teradata\TeradataQuote;
+use Throwable;
+
+final class DropWorkspaceHandler implements DriverCommandHandlerInterface
+{
+    private TeradataSessionManager $manager;
+
+    public function __construct(TeradataSessionManager $manager)
+    {
+        $this->manager = $manager;
+    }
+
+    /**
+     * @inheritDoc
+     * @param GenericBackendCredentials $credentials
+     * @param DropWorkspaceCommand $command
+     */
+    public function __invoke(
+        Message $credentials, // project credentials
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof DropWorkspaceCommand);
+
+        // validate
+        assert(!empty($command->getWorkspaceUserName()), 'DropWorkspaceCommand.workspaceUserName is required');
+        assert(!empty($command->getWorkspaceRoleName()), 'DropWorkspaceCommand.workspaceRoleName is required');
+        assert(!empty($command->getWorkspaceObjectName()), 'DropWorkspaceCommand.workspaceObjectName is required');
+
+        $db = $this->manager->createSession($credentials);
+
+        try {
+            $db->executeStatement(sprintf(
+                'DROP ROLE %s;',
+                TeradataQuote::quoteSingleIdentifier($command->getWorkspaceRoleName())
+            ));
+
+            $db->executeStatement(sprintf(
+                'DROP USER %s;',
+                TeradataQuote::quoteSingleIdentifier($command->getWorkspaceUserName())
+            ));
+        } catch (Throwable $e) {
+            if (!$command->getIgnoreErrors()) {
+                $db->close();
+                throw $e;
+            }
+        }
```

tady to imho není dobře, pokud máš ignore errors tak by to mělo zkusit zavolat oba DROPy takto když spadne první tak sa ten druhý nezavolá. A ten db close už není potřeba je to u DropBucket, protože ještě nebyl ten manager.

---

upraveno